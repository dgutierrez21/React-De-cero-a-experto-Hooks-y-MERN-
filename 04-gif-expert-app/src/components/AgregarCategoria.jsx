import { useState } from "react";

export const AgregarCategoria = ({ onNuevaCategoria }) => {
  const [valorInput, setValorInput] = useState("");

  const cambiarInput = (event) => {
    setValorInput(event.target.value);
  };

  const submit = (e) => {
    e.preventDefault();

    const valorInputSubmit = valorInput.trim();

    console.log(valorInput);

    if (valorInputSubmit.length < 2) {
      return;
    }

    // setCategorias((prevCategorias) => [valorInput, ...prevCategorias]);
    // el Hook useState devuelve una variable con el valor del estado actual y una función con la que puedes actualizar éste valor, la función siempre apuntará a la dirección de memoria de cualquier valor que le pasas, en el ejemplo el callback setCategories está apuntando a la dirección del último valor que le asignaste.

    onNuevaCategoria(valorInputSubmit);

    setValorInput("");
  };

  return (
    <form onSubmit={submit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={valorInput}
        onChange={cambiarInput}
      />
    </form>
  );
};
