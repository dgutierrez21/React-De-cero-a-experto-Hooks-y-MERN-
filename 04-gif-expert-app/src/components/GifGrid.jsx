import { useEffect } from "react";
import { useState } from "react";
import { getGifs } from "../helpers/GetGifs";

export const GifGrid = ({ categoria }) => {
  const [imagenes, setImagenes] = useState([]);

  // El gancho useEffect le permite realizar efectos secundarios en sus componentes.

  // Algunos ejemplos de efectos secundarios son: obtención de datos, actualización directa del DOM y temporizadores.

  // useEffect acepta dos argumentos. El segundo argumento es opcional.

  // useEffect(<función>, <dependencia>)

  const getImagenes = async () => {
    const nuevasImagenes = await getGifs(categoria);
    setImagenes(nuevasImagenes);
  };

  useEffect(() => {
    getImagenes();
  }, []);

  return (
    <>
      <h3>{categoria}</h3>

      <ol>
        {imagenes.map(({ id, titulo }) => (
          <li key={id}>{titulo}</li>
        ))}
      </ol>
    </>
  );
};
