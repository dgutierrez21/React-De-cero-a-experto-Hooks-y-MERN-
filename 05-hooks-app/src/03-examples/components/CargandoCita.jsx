export const CargandoCita = () => {
  return <div className="alert alert-info text-center">Cargando...</div>;
};
