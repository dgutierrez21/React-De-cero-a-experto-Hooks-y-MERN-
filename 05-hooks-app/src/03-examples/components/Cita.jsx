import { useState } from "react";
import { useRef } from "react";
import { useLayoutEffect } from "react";

export const Cita = ({ author, quote }) => {
  const pRef = useRef();

  const [boxSize, setBoxSize] = useState({ width: 0, height: 0 });

  // es parecido al useEffect a diferencia de que se ejecuta cuando todo se ha renderizado
  useLayoutEffect(() => {
    const { height, width } = pRef.current.getBoundingClientRect();

    setBoxSize({ height, width });
  }, []);
  return (
    <>
      <blockquote className="blockquote text-end" style={{ display: "flex" }}>
        <p className="mb-2" ref={pRef}>
          {quote}
        </p>
        <footer className="blockquote-footer">{author}</footer>
      </blockquote>
      <code> {JSON.stringify(boxSize)} </code>
    </>
  );
};
