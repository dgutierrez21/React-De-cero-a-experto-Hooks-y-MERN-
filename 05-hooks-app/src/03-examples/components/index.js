export * from "./CargandoCita";
export * from "./Cita";
