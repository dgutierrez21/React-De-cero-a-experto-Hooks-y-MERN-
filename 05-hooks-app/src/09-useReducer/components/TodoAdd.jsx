import { useForm } from "../../hooks";

export const TodoAdd = ({ onNewTodo }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: "",
  });

  const onFormSubmit = (e) => {
    e.preventDefault();

    if (description.length < 2) {
      return;
    }

    console.log(description.length);
    const newTodo = {
      id: new Date().getTime(),
      done: false,
      description: description,
    };

    onNewTodo(newTodo);
    onResetForm();
  };
  return (
    <form onSubmit={onFormSubmit}>
      <input
        type="text"
        placeholder="Que hay que hacer?"
        className="form-control"
        name="description"
        value={description}
        onChange={onInputChange}
      />

      <button type="submit" className="btn btn-outline-primary mt-2">
        Agregar
      </button>
    </form>
  );
};
