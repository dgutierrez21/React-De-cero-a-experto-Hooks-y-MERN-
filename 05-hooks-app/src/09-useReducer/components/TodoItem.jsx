export const TodoItem = ({
  todo: { description, id, done },
  onDeleteTodo,
  onToggleTodo,
}) => {
  return (
    <li className="list-group-item d-flex justify-content-between ">
      <span
        role="button"
        className={`align-self-center ${
          done ? "text-decoration-line-through" : ""
        }`}
        onClick={() => onToggleTodo(id)}
        data-testid="span"
      >
        {description}
      </span>
      <button
        className="btn btn-danger"
        onClick={() => onDeleteTodo(id)}
        data-testid="btnBorrar"
      >
        Borrar
      </button>
    </li>
  );
};
