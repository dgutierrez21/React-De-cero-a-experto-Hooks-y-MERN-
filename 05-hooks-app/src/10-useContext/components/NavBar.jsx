import React from "react";
import { Link } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="d-flex gap-3">
      <Link to={"/"}>Home</Link>
      <Link to={"about"}>About</Link>
      <Link to={"login"}>Login</Link>
    </nav>
  );
};
