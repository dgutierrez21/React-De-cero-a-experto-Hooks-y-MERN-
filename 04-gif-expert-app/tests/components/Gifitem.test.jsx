import { GifItem } from "../../src/components/GifItem";
import { render } from "@testing-library/react";

describe("Pruebas en el componente Gifitem", () => {
  const titulo = "Valorant",
    url = "https://afkasflkjasf.com";
  test("Evaluar snapshot", () => {
    const { container } = render(<GifItem titulo={titulo} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
