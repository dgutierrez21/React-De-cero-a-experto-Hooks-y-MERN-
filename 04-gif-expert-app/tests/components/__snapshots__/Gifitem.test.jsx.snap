// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Gifitem Evaluar snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Valorant"
      src="https://afkasflkjasf.com"
    />
    <p>
      Valorant
    </p>
  </div>
</div>
`;
